src/WebJobs.Extensions.CosmosDB/WebJobsConfigurationExtensions.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Azure.WebJobs.Extensions.CosmosDB
+{
+    internal static class WebJobsConfigurationExtensions
+    {
+        private const string WebJobsConfigurationSectionName = "AzureWebJobs";
+
+        public static IConfigurationSection GetWebJobsConnectionStringSection(this IConfiguration configuration, string connectionStringName)
```

Comment:
Can you use these helpers directly? https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Host/Extensions/IConfigurationExtensions.cs#L15

Reply:
Those return a string and we need an `IConfigurationSection` here.

Reply:
It might be nice if we added a  `IConfigurationSection`-returning overload to `IConfigurationExtensions`. Most extensions that do AAD auth use it.

Reply:
The problem is, `WebJobsConfigurationExtensions` resolve using the WebJobs root path `AzureWebJobs:extensions:<config name>`: https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Host/Extensions/IConfigurationExtensions.cs#L42-L46

The other method available `GetConnectionStringOrSetting` only returns a string, not the Section, so we cannot use it.

If we take a look at the other repository uses these methods, which are the ones I copied: https://github.com/Azure/azure-sdk-for-net/blob/303b75341ba17aecf239a3232d28ca76209a5346/sdk/extensions/Microsoft.Azure.WebJobs.Extensions.Clients/src/Shared/WebJobsConfigurationExtensions.cs#L12

Reply:
Yeah, I agree with you about the need for getting a section but you already have a copy of the WebJobsConfigurationExtensions.cs in this PR. Why not use it?

Reply:
Ah, I see now... makes sense...

Reply:
> Yeah, I agree with you about the need for getting a section but you already have a copy of the WebJobsConfigurationExtensions.cs in this PR. Why not use it?

Not sure I follow. I did create the copy (of the Azure SDK repo) and I am using it in the `DefaultCosmosDBServiceFactory.cs` file.